book-cw/part-2-coding/regex.adoc

```diff
@@ -1,3 +1,348 @@
 = Coding in Regex
 
-Chapter content removed for re-review (in progress).
+The patterning syntax known as _regular expressions_ or _regex_ is as notorious for its alleged steep learning curve as for its confirmed usefulness.
+Nobody seriously questions the _efficacy_ of this set of scripting rules, but the efficiency of writing regular expressions is sometimes unclear.
+Knowing when they're essential takes some familiarity, so we'll consider a few use=case types.
+
+Regular expressions can be used used not only to scan for predefined patterns in arbitrary text; expressions can extract contextualized data from within those matched patterns, passing it to an application for further manipulation and storage.
```

Comment:
"Not only can regular expressions search for patterns in text, but they can extract data from the matched patterns and pass that data to an application for further manipulation and storage."